src/jvm/test/org/jetbrains/kotlinx/lincheck_test/util/TestUtils.kt

```diff
@@ -22,43 +22,94 @@ import java.io.File
+    check(!expectedOutputFilePrefix.contains(".txt")) { "should only be prefix of outputfile" }
     check(this != null) { "The test should fail" }
-
     val actualOutput = StringBuilder().appendFailure(this).toString()
-    val expectedOutput = getExpectedLogFromResources(expectedOutputFile)
+    
+    // Always overwrite jdk8 non-trace
+    if (testJdkVersion == TestJdkVersion.JDK_8 && !isInTraceDebuggerMode && OVERWRITE_REPRESENTATION_TESTS_OUTPUT) {
+        getExpectedLogFileFromSources(getFileNameFor(expectedOutputFilePrefix, TestJdkVersion.JDK_8, false)).writeText(actualOutput)
+        return
+    }
+    
+    val compareToFile = getFileToCompareTo(expectedOutputFilePrefix)
+    val expectedOutput = getExpectedLogFromResources(compareToFile)
 
     if (actualOutput.filtered != expectedOutput.filtered) {
         if (OVERWRITE_REPRESENTATION_TESTS_OUTPUT) {
-            getExpectedLogFileFromSources(expectedOutputFile).writeText(actualOutput)
+            getExpectedLogFileFromSources(getFileNameFor(expectedOutputFilePrefix, testJdkVersion, isInTraceDebuggerMode)).writeText(actualOutput)
         } else {
             assertEquals(expectedOutput, actualOutput)
         }
     }
 }
 
+// To prevent file duplication this function finds the file to compare the results to.
+// With preference for jdk: 17, 15, 13, 11, 8 (trace mode) and 17, 15, 13, 11, 8 (non trace mode)
+// Search starts at current jdk level
+// For instance we are running tests for jdk 11 (trace mode), we will check file existence in the following order:
+// 11 (trace), 8 (trace), 11 (non-trace) and 8 (non-trace).
+private fun getFileToCompareTo(expectedOutputFilePrefix: String): String {
+    // If in trace mode first check if a trace debugger file can be found
+    if (isInTraceDebuggerMode) {
+        for (i in testJdkVersion.ordinal downTo 0) {
+            val jdkVersion = TestJdkVersion.entries[i]
+            val fileName = getFileNameFor(expectedOutputFilePrefix, jdkVersion, true)
+            if (logFileFromResourcesExists(fileName)) return fileName
+        }
+    }
+    // Check lower sdks in non-trace mode
+    for (i in testJdkVersion.ordinal downTo 0) {
+        val jdkVersion = TestJdkVersion.entries[i]
+        val fileName = getFileNameFor(expectedOutputFilePrefix, jdkVersion, false)
+        if (logFileFromResourcesExists(fileName)) return fileName
+    }
+    error("No file exists yet for this test and current jdk = $testJdkVersion, please run on jdk 8 with overwrite enabled")
+}
+
+// Generates file name in the form of 
+// prefix.txt, prefix_jdk_15.txt, prefix_trace_debugger.txt, prefix_trace_debugger_jdk_15.txt, etc..
+private fun getFileNameFor(expectedOutputFilePrefix: String, jdkVersion: TestJdkVersion, traceMode: Boolean): String {
+    return "${expectedOutputFilePrefix}${if (traceMode) "_trace_debugger" else ""}${
+        if (jdkVersion == TestJdkVersion.JDK_8) "" else "_${jdkVersion.name.lowercase()}"
```

Comment:
Instead of `${jdkVersion.name.lowercase()}` could you please override `toString()` of `TestJdkVersion`?
Also, please keep `_jdkN` format (without underscore).